src/report.rs

```diff
@@ -0,0 +1,376 @@
+            .expect("Failed to deserialize record");
+        assert_eq!(record2.id, 2);
+        assert_eq!(record2.value, "Value 2");
+
+        std::fs::remove_file(filename).expect("Couldn't delete file");
+    }
+
+    #[test]
+    fn multithreaded_report_generation_thread_local() {
+        use std::thread;
+
+        let num_threads = 10;
+        let num_reports_per_thread = 5;
+
+        let mut handles = vec![];
+
+        for i in 0..num_threads {
+            let handle = thread::spawn(move || {
+                let mut context = Context::new();
+                let config = context.get_config_options();
+                config
+                    .file_prefix("MultThreads".to_string())
+                    .directory("./".to_string());
+                context.add_report::<SampleReport>(&format!("thread_{}_SampleReport", i));
+
+                for j in 0..num_reports_per_thread {
+                    let report = SampleReport {
+                        id: (i * num_reports_per_thread + j) as u32,
+                        value: format!("Thread {} Report {}", i, j),
+                    };
+                    context.send_report(report);
+                }
+            });
+
+            handles.push(handle);
+        }
+
+        for handle in handles {
+            handle.join().expect("Thread failed");
+        }
+
+        for i in 0..num_threads {
+            let filename = format!("MultThreadsthread_{}_SampleReport.csv", i);
+            let path = std::path::Path::new(&filename);
+            assert!(path.exists(), "CSV file {} should exist", filename);
+
+            let mut reader = csv::Reader::from_path(path).expect("Failed to open CSV file");
+            let records = reader.deserialize::<SampleReport>();
+
+            for (j, record) in records.enumerate() {
+                let record: SampleReport = record.expect("Failed to deserialize record");
+                assert_eq!(record.id, (i * num_reports_per_thread + j) as u32);
+            }
+
+            std::fs::remove_file(filename).expect("Couldn't delete file");
+        }
+    }
+
+    #[derive(Serialize)]
+    struct TestStruct {
+        id: u32,
+        name: String,
+        active: bool,
+        category: TestEnum,
+    }
+
+    #[derive(Serialize)]
+    enum TestEnum {
+        Category1,
+        Category2,
+    }
+
+    #[test]
+    fn test_csv_serialization() -> Result<(), Box<dyn Error>> {
+        let mut writer = Writer::from_writer(vec![]);
+
+        let data = vec![
+            TestStruct {
+                id: 1,
+                name: "Test1".to_string(),
+                active: true,
+                category: TestEnum::Category1,
+            },
+            TestStruct {
+                id: 2,
+                name: "Test2".to_string(),
+                active: false,
+                category: TestEnum::Category2,
+            },
+        ];
+
+        for record in data {
+            writer.serialize(record)?;
```

Comment:
You should be testing the output of the library itself, not constructing a writer